### ISSUE TITLE
Origin header is undefined/empty when using app.inject

### ISSUE THREAD

**Issue by kamatheuska** [fastify/light-my-request#235]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.10.2

### Plugin version

5.8.0

### Node.js version

18.12.1

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

22.04

### Description

Seems that app.inject sends an empty Origin header which causes integration test to fail if fastify cors is settled.

If you think this could be a first good issue I could try to figure it out why that is the case and submit a PR

### Steps to Reproduce

Here is the test, notice the origin header. Without it, the test would fail with a 500 error, and every other test in the project as well.

https://github.com/kamatheuska/inventory-app/blob/master/packages/api/test/plugins/cors.test.ts#L10

### Expected Behavior

As a user, I expect `app.inject` to fake an origin header for me.

**Comment by mcollina** [comment]
Go for it, it's definitely a good first issue!

**Comment by alainrk** [comment]
Hi! I've tried to address this with a [small PR](https://github.com/fastify/light-my-request/pull/237), let me know if that could work for you!

**Comment by climba03003** [comment]
`Origin` header is not sent in all request in browser.
All the app should never assume `origin` header to be exist in all case.

Instead of adding it by default, we should provide a generator for default value.
So, the user do not need to pass the header one by one.

**Comment by alainrk** [comment]
Good point, according to https://fetch.spec.whatwg.org/#origin-header the logic to set the value is more articulated and based on the context as you mentioned

**Comment by mcollina** [comment]
After testing [this PR](https://github.com/fastify/light-my-request/pull/237) a bit, I think it's incorrect for this module to always add a `origin` header, because it would make it impossible to test the case where an `origin` header is not specified.

What I think we should be doing is to allow Fastify to set an `origin` header for all outgoing `inject()` request. Something like the following:

```js
const fastify = Fastify({
  injectOrigin: 'http://example.com:8080' // default: undefined
})
```

**Comment by mcollina** [comment]
I'm transfering it to the Fastify repository.
